src/test/javascript/spec/component/account/vcs-access-token-settings.component.spec.ts

```diff
@@ -0,0 +1,187 @@
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { VcsAccessTokensSettingsComponent } from 'app/shared/user-settings/vcs-access-tokens-settings/vcs-access-tokens-settings.component';
+import dayjs from 'dayjs/esm';
+import { ArtemisDatePipe } from 'app/shared/pipes/artemis-date.pipe';
+import { AlertService } from 'app/core/util/alert.service';
+
+describe('VcsAccessTokensSettingsComponent', () => {
+    let fixture: ComponentFixture<VcsAccessTokensSettingsComponent>;
+    let comp: VcsAccessTokensSettingsComponent;
+
+    let accountServiceMock: { getAuthenticationState: jest.Mock; deleteUserVcsAccessToken: jest.Mock; addNewVcsAccessToken: jest.Mock };
+    const alertServiceMock = { error: jest.fn(), addAlert: jest.fn() };
+    let translateService: TranslateService;
+
+    const token = 'initial-token';
+
+    beforeEach(async () => {
+        accountServiceMock = {
+            getAuthenticationState: jest.fn(),
+            deleteUserVcsAccessToken: jest.fn(),
+            addNewVcsAccessToken: jest.fn(),
+        };
+
+        await TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [
+                VcsAccessTokensSettingsComponent,
+                TranslatePipeMock,
+                MockPipe(ArtemisDatePipe),
+                MockComponent(ButtonComponent),
+                MockComponent(FormDateTimePickerComponent),
+            ],
+            providers: [
+                { provide: AccountService, useValue: accountServiceMock },
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: NgbModal, useClass: MockNgbModalService },
+                { provide: AlertService, useValue: alertServiceMock },
+            ],
+        }).compileComponents();
+        fixture = TestBed.createComponent(VcsAccessTokensSettingsComponent);
+        comp = fixture.componentInstance;
+
+        translateService = TestBed.inject(TranslateService);
+        translateService.currentLang = 'en';
+
+        accountServiceMock.getAuthenticationState.mockReturnValue(of({ id: 1, vcsAccessToken: token, vcsAccessTokenExpiryDate: '11:20' } as User));
+        accountServiceMock.addNewVcsAccessToken.mockReturnValue(of({ id: 1, vcsAccessToken: token, vcsAccessTokenExpiryDate: '11:20' } as User));
+        accountServiceMock.deleteUserVcsAccessToken.mockReturnValue(of({}));
+        jest.spyOn(console, 'error').mockImplementation(() => {});
+    });
+
+    it('should cancel token creation', () => {
+        accountServiceMock.getAuthenticationState.mockReturnValue(of({ id: 1 } as User));
+
+        startTokenCreation();
+
+        // click button to send expiry date to server, to create the new token
+        const createTokenButton = fixture.debugElement.query(By.css('#cancel-vcs-token-creation-button'));
+        createTokenButton.triggerEventHandler('onClick', null);
+        fixture.detectChanges();
+        expect(comp.edit).toBeFalsy();
```

Comment:
```suggestion
        expect(comp.edit).toBeFalse();
```
The `edit` attribute is a Boolean that cannot be `undefined`. Therefore, allowing it as part of the assertion does not make sense and is also discouraged by our client testing guidelines.

Same for `truthy()` in other tests below.

Reply:
True, that's cleaner :+1: